demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+                device_nstreams[device] = std::stoi(
+                    ie.GetConfig(device, CONFIG_KEY(CPU_THROUGHPUT_STREAMS)).as<std::string>());
+            } else if (device == ("GPU")) {
+                ie.SetConfig({{ CONFIG_KEY(GPU_THROUGHPUT_STREAMS),
+                                (device_nstreams.count(device) > 0 ? std::to_string(device_nstreams.at(device)) :
+                                                                         "GPU_THROUGHPUT_AUTO") }}, device);
+                device_nstreams[device] = std::stoi(
+                    ie.GetConfig(device, CONFIG_KEY(GPU_THROUGHPUT_STREAMS)).as<std::string>());
+
+                if ((FLAGS_d.find("MULTI") != std::string::npos) &&
+                    (FLAGS_d.find("CPU") != std::string::npos)) {
+                    // multi-device execution with the CPU + GPU performs best with GPU throttling hint,
+                    // which releases another CPU thread (that is otherwise used by the GPU driver for active polling)
+                    ie.SetConfig({{ CLDNN_CONFIG_KEY(PLUGIN_THROTTLE), "1" }}, "GPU");
+                }
+            } else if (device == "MYRIAD") {
+                ie.SetConfig({{ CONFIG_KEY(LOG_LEVEL), CONFIG_VALUE(LOG_NONE) },
+                              { VPU_CONFIG_KEY(LOG_LEVEL), CONFIG_VALUE(LOG_WARNING) }}, device);
+            }
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // --------------------------------------Load network to device---------------------------------------
+        ExecutableNetwork executableNetwork;
+        executableNetwork = ie.LoadNetwork(network, FLAGS_d);
+        // ---------------------------------------------------------------------------------------------------
+
+        // ----------------------------Try to set optimal number of infer requests----------------------------
+        if (FLAGS_nireq == 0) {
+            std::string key = METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS);
+            try {
+                FLAGS_nireq = executableNetwork.GetMetric(key).as<unsigned int>();
+            } catch (const details::InferenceEngineException& ex) {
+                THROW_IE_EXCEPTION
+                        << "Every device used with the imagenet_classification_demo should "
+                        << "support OPTIMAL_NUMBER_OF_INFER_REQUESTS ExecutableNetwork metric. "
+                        << "Failed to query the metric for the " << FLAGS_d << " with error:" << ex.what();
+            }
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // ---------------------------------------Create infer request----------------------------------------
+        std::vector<InferRequest> inferRequests;
+        int infReqNum = FLAGS_nireq;
+        for (int infReqID = 0; infReqID < infReqNum; ++infReqID) {
```

Comment:
`infReqNum` can be removed.
`FLAGS_nireq` is unsigned, so `infReqID` should also be unsigned.